bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacConnectorImpl.java

```diff
@@ -0,0 +1,481 @@
+        lockObj.lock();
+        try {
+            logger.debug("OnError");
+            completedRequest++;
+        } finally {
+            lockObj.unlock();
+        }
+
+        if (cb == null || request == null) {
+            return;
+        }
+
+        try {
+            final Request retryRequest = httpClient.newRequest(request.getURI());
+            request.method(HttpMethod.GET);
+
+            if (retryRequest != null) {
+                executeRequest(retryRequest, cb);
+            }
+        } catch (Exception ex) {
+            logger.error("exception");
+            throw ex;
+        }
+    }
+
+    private @Nullable ContentResponse executeRequest(final Request request, @Nullable SiemensHvacCallback callback)
+            throws Exception {
+        request.timeout(240, TimeUnit.SECONDS);
+
+        ContentResponse response = null;
+
+        @Nullable
+        SiemensHvacRequestListener requestListener = null;
+        if (callback != null) {
+            requestListener = new SiemensHvacRequestListener(callback, this);
+            request.onResponseSuccess(requestListener);
+            request.onResponseFailure(requestListener);
+        }
+
+        try {
+            if (requestListener != null) {
+                lockObj.lock();
+                try {
+                    startedRequest++;
+                    logger.debug("StartedRequest : {}", startedRequest);
+
+                } finally {
+                    lockObj.unlock();
+                }
+
+                request.send(requestListener);
+            } else {
+                response = request.send();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new SiemensHvacException("siemensHvac:Exception by executing request: " + request.getQuery() + " ; "
+                    + e.getLocalizedMessage());
+        }
+        return response;
+    }
+
+    private void _initConfig() throws Exception {
+
+        Configuration config = null;
+        SiemensHvacBridgeBaseThingHandler lcHvacBridgeBaseThingHandler = hvacBridgeBaseThingHandler;
+
+        if (lcHvacBridgeBaseThingHandler != null) {
+            Bridge bridge = lcHvacBridgeBaseThingHandler.getThing();
+            config = bridge.getConfiguration();
+        } else {
+            throw new SiemensHvacException(
+                    "siemensHvac:Exception unable to get config because hvacBridgeBaseThingHandler is null");
+        }
+
+        if (config.containsKey("baseUrl")) {
+            baseUrl = (String) config.get("baseUrl");
+        }
+
+        if (config.containsKey("userName")) {
+            userName = (String) config.get("userName");
+        }
+        if (config.containsKey("userPassword")) {
+            userPassword = (String) config.get("userPassword");
+        }
+    }
+
+    private void _doAuth(boolean http) throws Exception {
```

Comment:
Same here

Reply:
I suppose your concern is about the underscore ?
Remove it.